ProcessLib/LIE/HydroMechanics/LocalAssembler/IntegrationPointDataFracture.h

```diff
@@ -0,0 +1,79 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESSLIB_LIE_HYDROMECHANICS_INTEGRATIONPOINTDATAFRACTURE_H_
+#define PROCESSLIB_LIE_HYDROMECHANICS_INTEGRATIONPOINTDATAFRACTURE_H_
+
+#include <Eigen/Eigen>
+
+#include "MaterialLib/FractureModels/FractureModelBase.h"
+
+namespace ProcessLib
+{
+namespace LIE
+{
+namespace HydroMechanics
+{
+template <typename HMatricesType, typename ShapeMatrixTypeDisplacement,
+          typename ShapeMatrixTypePressure, unsigned GlobalDim>
+struct IntegrationPointDataFracture final
+{
+    explicit IntegrationPointDataFracture(
+        MaterialLib::Fracture::FractureModelBase<GlobalDim>& fracture_material_)
+        : fracture_material(fracture_material_)
+    {
+    }
+
+    IntegrationPointDataFracture(IntegrationPointDataFracture&& other)
+        : H_u(std::move(other.H_u)),
+          sigma_eff(std::move(other.sigma_eff)),
+          sigma_eff_prev(std::move(other.sigma_eff_prev)),
+          w(std::move(other.w)),
+          w_prev(std::move(other.w_prev)),
+          N_p(std::move(other.N_p)),
+          dNdx_p(std::move(other.dNdx_p)),
+          aperture(std::move(other.aperture)),
+          aperture0(std::move(other.aperture0)),
+          fracture_material(other.fracture_material),
+          C(std::move(other.C)),
+          integration_weight(std::move(other.integration_weight)),
+          darcy_velocity(std::move(other.darcy_velocity))
+    {
+    }
+
+    typename HMatricesType::HMatrixType H_u;
```

Comment:
Note: Default-constructing these matrices might fail miserably if they are of dynamic size. Currently that probably doesn't matter, because displacement-related stuff is always fixed size.